web/source/osk/oskBaseKey.ts

```diff
@@ -79,56 +81,21 @@ namespace com.keyman.osk {
       btn.appendChild(skIcon);
     }
 
-    construct(vkbd: VisualKeyboard, displayUnderlying: boolean, rowStyle: CSSStyleDeclaration, totalPercent: number): {element: HTMLDivElement, percent: number} {
+    construct(vkbd: VisualKeyboard): HTMLDivElement {
       let spec = this.spec;
-      let isDesktop = vkbd.device.formFactor == "desktop"
 
       let kDiv = document.createElement('div');
       kDiv.className='kmw-key-square';
 
-      let ks=kDiv.style;
-      ks.width=this.objectGeometry(vkbd, spec['widthpc']);
```

Comment:
Much of the code removed here was relocated to the class's `refreshLayout` method.